src/circuit/gadgets/constraints.rs

```diff
@@ -71,6 +111,39 @@ pub fn add<F: PrimeField, CS: ConstraintSystem<F>>(
     Ok(res)
 }
 
+/// Adds a constraint to CS, enforcing that the addition of the allocated booleans,
+/// interpreted as nums, in vector `v` is equal to `sum`.
+///
+/// summation(v) = sum
+pub fn boolean_summation<F: PrimeField, CS: ConstraintSystem<F>>(
```

Comment:
Can you add unit tests for this function please?

Reply:
added